storage-s3-dynamodb/src/main/java/io/delta/storage/BaseExternalLogStore.java

```diff
@@ -311,20 +358,23 @@ private void fixDeltaLog(FileSystem fs, ExternalCommitEntry entry) throws IOExce
     }
 
    /**
-    * Copies file within filesystem
+    * Copies file within filesystem.
+    *
     * @param fs reference to [[FileSystem]]
     * @param src path to source file
     * @param dst path to destination file
     */
     private void copyFile(FileSystem fs, Path src, Path dst) throws IOException {
         LOG.debug("copy file: {} -> {}", src, dst);
-        FSDataInputStream input_stream = fs.open(src);
-        FSDataOutputStream output_stream = fs.create(dst, true);
+        final FSDataInputStream inputStream = fs.open(src);
         try {
-            IOUtils.copy(input_stream, output_stream);
-            output_stream.close();
+            final FSDataOutputStream outputStream = fs.create(dst, false); // overwrite=false
+            IOUtils.copy(inputStream, outputStream);
+            outputStream.close();
+        } catch (org.apache.hadoop.fs.FileAlreadyExistsException e) {
+            throw new java.nio.file.FileAlreadyExistsException(dst.toString());
```

Comment:
I realized that whether changing `overwrite` to `false` doesn't matter for correctness because `copyFile` is called from two methods `fixDeltaLog` and `write`, and both of them will ignore any `Throwable`.

There are two cases we may hit `FileAlreadyExistsException` here:

1. `fixDeltaLog` is called at the same time when a write is completing the write. `FileAlreadyExistsException` will be ignored by `fixDeltaLog` properly. The writer side should also ignore it. I guess this is why it's `overwrite=true` previously.
2. `expirationDelaySeconds` is set to a small value such as 0. In this case, we need to notify the writer that this file does exist.

It seems we don't have any information in the writer side to distinguish these two cases. However, the default `expirationDelaySeconds` is 1 day which should be long enough to prevent Case 2.

Changing it to `false` may be still helpful for performance because we don't need to upload the file if the file already exists.

In a second thought, maybe we should ignore `FileAlreadyExistsException` here? Otherwise, `fixDeltaLog` and `write` may log a lot of `FileAlreadyExistsException`s, which is confusing.

Reply:
> In a second thought, maybe we should ignore FileAlreadyExistsException here? Otherwise, fixDeltaLog and write may log a lot of FileAlreadyExistsExceptions, which is confusing.

both of those APIs will first check if the destination file exists first before trying to copy. so do you really think they may log a lot of `FileAlreadyExistsExceptions`?

Reply:
That's fair